correct_periscope_drift.py

```diff
@@ -0,0 +1,421 @@
+    #
+    return mypars
+
+
+# Display parameter info to the user.
+# The format and choice of information to display is up to you;
+# I've chosen to include some ancilalry information but it
+# depends on the tool what should be used.
+#
+def display_start_info(opts):
+    v1("Running: {0}".format(opts["progname"]))
+    v2("  version = {0}".format(VERSION))
+    v2("with parameters:")
+    v2("  infile={0}".format(opts["infile"]))
+    v2("  evtfile={0}".format(opts["evtfile"]))
+    v2("  outfile={0}".format(opts["outfile"]))
+    # probably other values here too
+    v2("  verbose={0}".format(opts["verbose"]))
+    v2("  and ASCDS_INSTALL is {0}".format(os.environ["ASCDS_INSTALL"]))
+    v2("-" * 60)
+
+
+def equatorial2transform(ra, dec, roll):
+    """Construct the transform/rotation matrix from RA,Dec,Roll (given in degrees)
+    :returns: transform matrix
+    :rtype: Nx3x3 numpy array
+
+    """
+    ra = np.radians(ra)
+    dec = np.radians(dec)
+    roll = np.radians(roll)
+    ca = np.cos(ra)
+    sa = np.sin(ra)
+    cd = np.cos(dec)
+    sd = np.sin(dec)
+    cr = np.cos(roll)
+    sr = np.sin(roll)
+    # This is the transpose of the transformation matrix (related to
+    # translation of original perl code
+    rmat = np.array(
+        [[ca * cd,                  sa * cd,                sd     ],
+         [-ca * sd * sr - sa * cr, -sa * sd * sr + ca * cr, cd * sr],
+         [-ca * sd * cr + sa * sr, -sa * sd * cr - ca * sr, cd * cr]])
+
+    return rmat.transpose()
+
+
+def radec2eci(ra, dec):
+    """
+    Convert from RA,Dec to ECI.  The input ``ra`` and ``dec`` values can be 1-d
+    arrays of length N in which case the output ``ECI`` will be an array with
+    shape (3,N).
+
+    Borrowed from Ska.quatutil
+
+    :param ra: Right Ascension (degrees)
+    :param dec: Declination (degrees)
+    :returns: numpy array ECI (3-vector or 3xN array)
+    """
+    r = np.radians(ra)
+    d = np.radians(dec)
+    return np.array([np.cos(r) * np.cos(d), np.sin(r) * np.cos(d), np.sin(d)])
+
+
+#def roll_range(asp_roll):
+#    max_roll = np.max(asp_roll)
+#    min_roll = np.min(asp_roll)
+#    if (max_roll - min_roll) > 180:
+#        roll_180 = asp_roll.copy()
+#        roll_180[roll_180 > 180] -= 360
+#        max_roll = np.max(roll_180)
+#        min_roll = np.min(roll_180)
+#        if max_roll - min_roll > 180:
+#            raise ValueError("> 180 deg roll range")
+#    return max_roll - min_roll
+#
+
+def extract_events(event_file, src_x, src_y, src_radius):
+    """
+    Get events from specified source circle
+
+    :param event_file: Chandra event 1 or 2 file
+    :param src_x: Sky X coordinate of source region to extract
+    :param src_y: Sky Y coordinate of source region to extract
+    :param src_radius: Source region/circle radius in pixels
+    :returns: CRATE of events
+    """
+    regstring = "circle({},{},{})".format(src_x, src_y, src_radius)
+    events = pycrates.read_file("{}[sky={}]".format(
```

Comment:
Kenny's comment was not addressed (but marked as outdated by context in the PR) so pasteing it in again:

Two things

1. It looks like from here, the `radius` is going to be in physical pixels, rather than in arcsec which is what the .par file says.
2. We should really double check the ra/dec values.  We have in the past seen bugs in crates/DM where filtering in `sky` but then accessing the virtual WCS columns `ra` and `dec` produced incorrect results.  I'll need to look this up unless @DougBurke remembers this more.